geode-core/src/main/java/org/apache/geode/management/internal/cli/result/ResultBuilder.java

```diff
@@ -188,6 +189,22 @@ public static CommandResult buildResult(ResultData resultData) {
     return new CommandResult(resultData);
   }
 
+  public static CommandResult buildResult(List<CliFunctionResult> functionResults) {
+    TabularResultData tabularData = ResultBuilder.createTabularResultData();
+    boolean success = false;
+    for (CliFunctionResult result : functionResults) {
+      tabularData.accumulate("Member", result.getMemberIdOrName());
+      tabularData.accumulate("Status", result.getStatus());
+      // if one member returns back successful results, the command results in success
+      if (result.isSuccessful()) {
+        success = true;
+      }
+    }
+
+    tabularData.setStatus(success ? Result.Status.OK : Result.Status.ERROR);
+    return ResultBuilder.buildResult(tabularData);
+  }
```

Comment:
There are two ways of determining command success, 1) If function succeeds on (at least) one member, then the command succeeds, 2) If function fails on (at least) one member, then the command fails.

This implementation only supports case (1).

Reply:
right, as far as I see, this is the only case. (destroy region, create region, create index .....). I think this promotes consistency.

Reply:
That doesn't really make sense to me, though.  If a region should be destroyed across multiple members, and that destruction fails to complete on at least one member, then that region has not been destroyed.  In my mind, a failure on any member is a failure of the command as a whole.

Looking at some of the test changes from expecting `statusIsError` to `statusIsSuccess`, it seems like that was the established behavior (unless I'm misreading the old tests).

Reply:
region.destroy is a distributed action, unless scope is LOCAL. So a replicated/partitioned region, if destroy is successful on one member, then it's destroyed in the entire cluster. And if there is one successful result, we would get xmlEntity to update the cluster config. That alone would also make the command deemed successful.

Reply:
Before this change, there is a lot of differences on how to report these FunctionResults back from different members. Different command handles it differently. Some would ignore failures, even if all results came back as failed, it would still mark the command as successful. Some would mark the command as failed, but if one result came back with a valid xml, it will still use it to update cluster config. So hopefully, with this change, we can have some consistency. Even that means we will need to change how command report failure/success

Reply:
Ahh, that makes sense.  And the same holds for the indexing, etc?

Reply:
With regard to destroy region and create region commands. You're assuming that when destroying, if destroy succeeded on one, say, member1, then it may fail on member2 because with a distributed destroy the region may already have been destroyed before the function is executed on member2.

However with create, the region is being created on a set of members, if creation succeeds on one member but fails on another, did the command really succeed?

Reply:
yes, if the region is created on at least one member, because cluster config will be updated with this region, so when a new server comes up, it will have that region on the new server. That aspect makes me think this command should be deemed successful.

Reply:
Also, often times, the creation of a region (or any entity) would fail on only partial members because that member already has that region, so you will see results like:
Member | Status
server-0 | regionA successfully created
server-1 | ERROR: failed to create the region because regionA already exists

In this case, the command should still be deemed successful.

Reply:
Yes ..., maybe ...? for instance if a replicated region is being created on a server group, but failed on one member in the group, then UNTIL a new server comes up the replicated copies may not exist. Should/does a user get warned about this condition? The result of the command isn't what the user expected.

However if we take the fact of the updated cluster config as the "truth", then your reasoning makes sense.

Reply:
deemed successful only means the command result status is OK, not ERROR. User does get warned about one member's region creation failed by the tabular view.

Reply:
OK. Does still bother me to see "ERROR" in a command output with a command status OK.

Reply:
I'm not sure we can come up with and cover all situations of partial failures for every command. I really like that we're being consistent across all commands (even if it ultimately turns out to be wrong; which I don't think it will). At the very least we're not hiding errors and now, with commands being idempotent, users should be able to recover from almost anything.